### PR TITLE
feat: add password input

### PR DESCRIPTION
**Pull request by maicongodinho**
Closes carbon-design-system/carbon-components-angular#1930

This update introduces a new password input component, encompassing all essential features available in the base text input component.

### CODE REVIEW

**Comment by maicongodinho** [comment]
In the context of React components, where a "directive-like" solution is not employed, my approach to the new component involves maintaining the use of projection. A type flag is utilized to seamlessly switch between templates representing different input types, thus ensuring native HTML5 password handling.

To implement this, users are required to utilize two inputs accompanied by the `cdsPassword` directive, each configured with distinct types (text and password). The `cds-password-label` is then responsible for managing these inputs effectively.

Building upon the React component reference found at https://react.carbondesignsystem.com/?path=/story/components-textinput--toggle-password-visibility:

**Comment by hubertmalkowski** [comment]
Sorry to butt in, butt is there any update on this PR?

**Comment by Akshat55** [comment]
Sorry, been preoccupied, let me prioritize this for this week and we can hopefully merge it sometimes next week. @maicongodinho, I'm going to make a few simplifications to this. I want to be able to toggle input property type so users don't have to project both `input[type="text"]` and `input[type="password"]`.

**Comment by maicongodinho** [comment]
Hey @Akshat55, indeed, this is an older implementation. I aimed to leverage the directive style, integrating best practices in security and accessibility from the React library. While exploring simplification within our existing structure, I couldn't find an optimal solution. However, we could consider developing a new, consolidated component that combines the functionality of labels and password features seamlessly. Let me know what you think and what you need from my side.

**Comment by maicongodinho** [comment]
Hi @Akshat55, I just saw the approach you took. Excellent use of setting the property to the native element. I hadn't considered this method myself. Really good work!

File: src/input/password-input-label.component.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	template: `
+		<label
+			[for]="labelInputID"
+			[attr.aria-label]="ariaLabel"
+			class="cds--label"
+			[ngClass]="{
+				'cds--label--disabled': disabled,
+				'cds--skeleton': skeleton
+			}"
+		>
```
Comment (Akshat55) [inline src/input/password-input-label.component.ts:42]:
```suggestion
			}">
```

File: src/input/password-input-label.component.ts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+				<ng-template *ngIf="isTemplate(warnText)" [ngTemplateOutlet]="warnText"></ng-template>
+			</div>
+		</div>
+    `
+})
+/**
+ * Represents the Password Input Label Component.
+ */
+export class PasswordInputLabelComponent extends BaseIconButton implements AfterViewInit {
+
```
Comment (Akshat55) [inline src/input/password-input-label.component.ts:119]:
```suggestion
```

**Review by Akshat55 — CHANGES_REQUESTED** [review]

**Review by Akshat55 — APPROVED** [review]

File: src/input/password.spec.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import { PasswordInputLabelComponent, PasswordInput } from "./";
+import { Tooltip } from "../tooltip";
+
+@Component({
+	template: `
+		<cds-password-label>
+			Password
+			<input
+				cdsPassword
+				type="password">
```
Comment (zvonimirfras) [inline src/input/password.spec.ts:14]:
```suggestion
			<input cdsPassword type="password">
```

**Review by zvonimirfras — APPROVED** [review]

**Review by Akshat55 — APPROVED** [review]

**Pull request merged**